runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/config/TlsVersion.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.config
+
+/**
+ * Describes a version of TLS
+ */
+@Suppress("ktlint:enum-entry-name-case")
+public enum class TlsVersion {
+    /**
+     * TLS version 1
+     */
+    Tls1_0,
```

Comment:
It appears we're inconsistent between SCREAMING_CASE (e.g., [`AlpnId`](https://github.com/awslabs/smithy-kotlin/blob/cd5759e4a34439c873863160179a0018e87cb6d9/runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/HttpClientEngineConfig.kt#L155-L176), [`PresigningLocation`](https://github.com/awslabs/smithy-kotlin/blob/cd5759e4a34439c873863160179a0018e87cb6d9/runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/Presigner.kt#L75-L85), [`TimestampFormat`](https://github.com/awslabs/smithy-kotlin/blob/cd5759e4a34439c873863160179a0018e87cb6d9/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/time/TimestampFormat.kt#L10-L40), etc.) and PascalCase (e.g., [`ErrorType`](https://github.com/awslabs/smithy-kotlin/blob/cd5759e4a34439c873863160179a0018e87cb6d9/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/Exceptions.kt#L138-L142), [`Phase`](https://github.com/awslabs/smithy-kotlin/blob/cd5759e4a34439c873863160179a0018e87cb6d9/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/io/middleware/Phase.kt#L19-L21), [`OsFamily`](https://github.com/awslabs/smithy-kotlin/blob/cd5759e4a34439c873863160179a0018e87cb6d9/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/Platform.kt#L36-L43), etc.). Interestingly, even [Kotlin's own style guide allows either "depending on usage"](https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties:~:text=For%20enum%20constants%2C%20it%27s%20OK%20to%20use%20either%20uppercase%20underscore%2Dseparated%20names%20(screaming%20snake%20case)%20(enum%20class%20Color%20%7B%20RED%2C%20GREEN%20%7D)%20or%20upper%20camel%20case%20names%2C%20depending%20on%20the%20usage.). I can switch this to SCREAMING_CASE since that matches [Kotlin's documented examples](https://kotlinlang.org/docs/enum-classes.html) but I wonder if we need to be more consistent throughout our project.

Reply:
I'd say we probably should be. Probably even worth a lint 🤔

Reply:
Yeah, agreed on lint rules for whatever we decide. The default **ktlint** ruleset already verifies enum names although [it allows both styles permitted by JB's Kotlin guidelines](https://github.com/pinterest/ktlint/blob/d195ffde7d5160616900d4787f92f4d08e6e1aa4/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/EnumEntryNameCaseRule.kt#L15). You may notice that I suppressed this rule on the enum since I wrote it in PascalCase but needed `_` as a digit separator. Maybe I should've gone for `Tls1Dot1` (/s).